src/ami-set.js

```diff
@@ -8,16 +8,37 @@ const KEY_CONST = 'ami-set';
  * by its AWS region.
  */
 
-let amiSet = base.Entity.configure({
+let AmiSet = base.Entity.configure({
   version: 1,
 
   partitionKey: base.Entity.keys.ConstantKey(KEY_CONST),
-  rowKey: base.Entity.keys.StringKey('amiSetId'),
+  rowKey: base.Entity.keys.StringKey('id'),
 
   properties: {
 
-    amiSetId: base.Entity.types.String,
+    id: base.Entity.types.String,
+    /* This is a JSON object which contains the AMIs of an AMI set keyed by
+     * their virtualization type and region. It is in the shape:
+     * {
```

Comment:
I tend to agree regarding the list of objects (the second option).  That will also mean we don't need to change our schema when we add a new region.  I also agree that we don't need to solve it in this patch.